src/components/user-list.js

```diff
@@ -15,51 +18,82 @@ export default class UserList extends React.Component {
 
     _getAppState () {
         return {
-            duplicateUserId: '',
-            wantAdd:         false,
-            users:           UserStore.getUsers()
+            duplicateUserId:   '',
+            highlightedFields: '',
+            addUserMode:       false,
+            searchMode:        false,
+            users:             UserStore.getUsers()
         };
     }
 
     _onChange = () => {
         this.setState(this._getAppState());
     };
 
-    _handleClickAddUser = () => {
-        this.setState({ wantAdd: !this.state.wantAdd });
+    _handleAddUserClick = () => {
+        this.setState({ addUserMode: !this.state.addUserMode });
     };
 
-    _addingFailed = userId => {
-        this.setState({ duplicateUserId: userId});
+    _addingFailed = (userId) => {
+        this.setState({ duplicateUserId: userId });
+    };
+
+    _handleFindUserClick = (e) => {
+        const { searchMode } = this.state;
+
+        this.setState({ searchMode: !searchMode });
+        e.preventDefault();
+    };
+
+    _usersFound = (usersFound, highlightedFields) => {
+        this.setState({ users: usersFound, highlightedFields: highlightedFields, addUserMode: false });
```

Comment:
(minor) can be simplified 
```suggestion
        this.setState({ users: usersFound, highlightedFields, addUserMode: false });
```